### PR TITLE
chore(rootfs-image): Add bootloader canary check.

### PR DESCRIPTION
**Pull request by kacf**
Ticket: MEN-6833

### CODE REVIEW

File: support/modules/rootfs-image [hunk 1]
```diff
@@ -81,6 +81,27 @@ set_upgrade_vars() {
...
 
+check_environment_canary() {
+    mender_check_saveenv_canary="$(${PRINTENV} mender_check_saveenv_canary)"
+    if [ "$mender_check_saveenv_canary" = "mender_check_saveenv_canary=1" ]; then
+	# If the check canary exists (added during build), we need to check the real canary to make
+	# sure that the boot loader was successful in adding it during boot.
+	mender_saveenv_canary="$(${PRINTENV} mender_saveenv_canary)"
+	if [ "$mender_saveenv_canary" != "mender_saveenv_canary=1" ]; then
+	    cat 1>&2 <<'EOF'
+`mender_check_saveenv_canary` was set in the boot environment, but
```
Comment (oleorhagen) [inline support/modules/rootfs-image:92]: Careful on the backquotes

File: support/modules/rootfs-image [hunk 2]
```diff
@@ -101,6 +122,7 @@ ensure_correct_root_mounted() {
 
 check_requirements() {
     parse_conf_file
+    check_environment_canary
```
Comment (oleorhagen) [inline support/modules/rootfs-image:125]: We should only warn, no hard exits?

File: support/modules/rootfs-image (hunk 1 above)
Comment (kacf) [reply support/modules/rootfs-image:92]: I already got burned! 😄

That's why `'EOF'` is single-quoted, it disables all expansions.

File: support/modules/rootfs-image (hunk 2 above)
Comment (kacf) [reply support/modules/rootfs-image:125]: `set -e` is set, so this is a hard exit.

File: support/modules/rootfs-image (hunk 1 above)
Comment (oleorhagen) [reply support/modules/rootfs-image:92]: haha, I figured. 

I would still suggest single-quoting this instead though, and remove the verbatim expansion. but YMMV

Comment (kacf) [reply support/modules/rootfs-image:92]: I don't think I want to introduce backslashes if we can avoid it, and we don't need expansions here, so I think I'll leave it as is.

**Review by oleorhagen — APPROVED** [review]

**Review by lluiscampos — COMMENTED** [review]
:heavy_check_mark: Looks good

**Pull request merged**